### ISSUE TITLE
ActiveModel#added? handles string attributes incorrectly when using a symbol message

### ISSUE THREAD

**Issue by jhubert** [rails/rails#33357]
### Steps to reproduce

Passing test case with activerecord 5.1.6:
```ruby
# frozen_string_literal: true

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "http://rubygems.org"

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "5.1.6"
end

require "active_record"
require "minitest/autorun"
require "logger"

class Comment
  include ActiveModel::Validations

  attr_accessor :author

  validates :author, presence: true
end

class BugTest < Minitest::Test
  def test_association_stuff
    comment = Comment.new
    comment.valid?

    assert comment.errors.added?(:author, :blank)
    assert comment.errors.added?("author", :blank)
  end
end
```

Failing test case with activerecord 5.2.0:
```ruby
# frozen_string_literal: true

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "http://rubygems.org"

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "5.2.0"
end

require "active_record"
require "minitest/autorun"
require "logger"

class Comment
  include ActiveModel::Validations

  attr_accessor :author

  validates :author, presence: true
end

class BugTest < Minitest::Test
  def test_association_stuff
    comment = Comment.new
    comment.valid?

    assert comment.errors.added?(:author, :blank)
    assert comment.errors.added?("author", :blank)
  end
end
```

### Expected behavior
Checking for an error on the attribute via added? should be possible with a string or a symbol as the attribute name.

### Actual behavior
In ActiveRecord 5.2.0 it's only possible to use a symbol. This is due to [a commit that was merged back in November](https://github.com/rails/rails/pull/31117/commits/15cb4efadb61a8813967d3c25f4adfc9a918a0c0). 

More information about this bug can be found [on the pull request](https://github.com/rails/rails/pull/31117).

### System configuration
**Rails version**: 5.2.0

**Ruby version**: 2.5.1

**Comment by twitnithegirl** [comment]
I was able to reproduce this bug locally. I also ran it against current master to check that no one else has fixed it recently.

**Comment by jhubert** [comment]
@twitnithegirl Great. I've submitted [a PR to fix it](https://github.com/rails/rails/pull/33361).

**Comment by Edouard-chin** [comment]
Fixed by #33361 we can close this :)
